### PR TITLE
Add prepend.php to set global settings from environment variables

### PR DESCRIPTION
**Pull request by beeradb**
## The Problem:
Pantheon commits the settings.php file for Drupal distros. We need to inject things like DB credentials and other site settings, but we can't do that in settings.php without modifying files which are controlled by git. 

## The Fix:
This approach allows us to get around that by using the PHP "auto_prepend_file" directive to set required settings without mucking with things that are controlled via git or belong to users (settings.local.php).

This is very similar to the approach used by Kalabox for controlling this information. https://github.com/kalabox/kalabox-app-pantheon/tree/v2.1/app/config/php
## Related Issue Link(s):
This is a part of the larger effort for pantheon integration going on at https://github.com/drud/ddev/pull/345 and it should not be merged quite yet.

### CODE REVIEW

**Comment by rickmanelius** [comment]
Naive question. I'm assuming that this doesn't preclude someone from working on two sites at the same time because we can only use one set of environment variables?

My initial reaction to this isn't particularly positive, but I realize we're trying to play nice with hard coded assumptions from repos generated directly or required by external hosting providers. In short, this allows us to achieve the objective of not requiring unnecessary manipulations to a committed settings.php file. My first question is whether or not a Drupal dev working off the command line and used to simply viewing the settings.php file will have a reasonably easy time understanding what we're doing when debugging. If we have an answer for that, I think that would alleviate most of my apprehension.

**Comment by beeradb** [comment]
No, this doesn't preclude someone from working on two sites at the same time. Each site has it's own containers, which each have their own set of environment variables. This doesn't affect (either positively or negatively) the ability to work on any number of sites.

With regards to developer confusion: All of these values end up in $_ENV and/or $_SERVER, and there's no special magic happening beyond getting them there. It's think an advanced PHP dev would be capable of finding how environment variables are getting placed in a php-fpm config, but that's just conjecture. Additionally, the code only executes when the provider is set to pantheon and there is nothing preventing users from modifying the values in their own code.

I do agree that the method isn't the first solution I would reach for, but I'm not aware of a better solution given the constraints.

**Comment by beeradb** [comment]
I also updated this PR to include a `/etc/drush/drushrc.php` file. This should allow drush to know the site URL, meaning commands such as `drush uli` will use a proper URL on D8!!

**Comment by rickmanelius** [comment]
Adding @rfay as a reviewer.

File: files/usr/share/php/prepend.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    */
+    if ((gettype($dbh->exec("SELECT count(*) FROM users")) == 'integer') != 1) {
+        $_SERVER['PANTHEON_DATABASE_STATE'] = 'empty';
+    }
+
+    // And now we're done; close it up!
+    $dbh = null;
+
+    }
+}
```
Comment (rfay) [inline files/usr/share/php/prepend.php]: Oh no!

**Comment by beeradb** [comment]
settings.local.php is a user controlled file that may contain any number of important overrides. I don't think it's very wise for us to attempt to inject anything there. At least I would find that very surprising and off-putting.

**Comment by rfay** [comment]
I understand your reservation here. However, this is exactly how settings.local.php is typically used, as a site-specific add-on to settings.php. And we could manage it in the same way we're currently managing settings.php, detecting whether we've created it.

**Comment by beeradb** [comment]
Closing this in favor of a settings.local.php / wp-local.php approach. There is some value here in the `drushrc.php` additions. so I'll open a followup with those.

**Pull request closed without merging**